hermes/bin/src/runtime_extensions/hermes/http_gateway/routing.rs

```diff
@@ -115,20 +115,39 @@ pub(crate) async fn router(
     Ok(response)
 }
 
-/// Route single request to hermes backend
+/// Routes HTTP requests to WASM modules or static file handlers
+///
+/// Converts incoming HTTP requests into structured events for WASM processing,
+/// preserving full URLs including query parameters for accurate forwarding.
+///
+/// ## Routing
+/// - `/api/*` → WASM modules via event queue
+/// - Valid paths → Static file system
+/// - Invalid paths → HTTP 404
+///
+/// ## Key Features
+/// - Preserves query parameters (e.g., `?asat=SLOT:95022059`)
+/// - Multi-value header support
+/// - Async request/response via MPSC channels
 async fn route_to_hermes(
     req: Request<Incoming>,
     app_name: ApplicationName,
 ) -> anyhow::Result<Response<Full<Bytes>>> {
+    // Create MPSC channel for async WASM communication
     let (lambda_send, lambda_recv_answer): (Sender<HTTPEventMsg>, Receiver<HTTPEventMsg>) =
         channel();
```

Comment:
I know it's not a part of this PR, but I'm wondering if a `oneshot` channel would suffice here?

Reply:
Good point! A oneshot channel would actually be more appropriate here. Looking at the usage pattern, we're creating a new channel for each HTTP request and only expect a single response back from the WASM module. The current MPSC channel is overkill since we never send multiple messages.
We'll address this in a follow-up PR to keep the scope of this change focused.